components/raftstore/src/store/entry_storage.rs

```diff
@@ -583,14 +583,18 @@ impl CacheWarmupState {
         self.started_at.saturating_elapsed()
     }
 
-    /// Whenther the warmup task is timeout.
+    /// Whether the warmup task is already timeout.
     pub fn is_task_timeout(&self) -> bool {
         self.is_task_timeout
     }
 
-    pub fn mark_task_timeout(&mut self) {
-        WARM_UP_ENTRY_CACHE_COUNTER.timeout.inc();
-        self.is_task_timeout = true;
+    /// Check if the task is timeout.
+    pub fn check_task_timeout(&mut self, duration: Duration) -> bool {
+        if self.elapsed() > duration {
+            WARM_UP_ENTRY_CACHE_COUNTER.timeout.inc();
+            self.is_task_timeout = true;
+        }
+        self.is_task_timeout
     }
 
     /// Whether the state is stale.
```

Comment:
use `check_task_timeout` to replace `is_stale` as well

Reply:
`check_stale` is added